lxd/instance.go

```diff
@@ -540,6 +541,9 @@ func autoCreateContainerSnapshots(ctx context.Context, d *Daemon, instances []in
 	return nil
 }
 
+var snapshotsInProgress = map[int]struct{}{}
```

Comment:
Bit of a misleading name, as the snapshots are being removed, not created.
How about `snapshotPruneRunning`, and then the associated mutex can be called `snapshotPruneRunningMu` to show its association to `snapshotPruneRunning`

Reply:
I've added new `PruneLock` struct and I've changed this part of code a little bit. Look at the comment below.